js/src/jupyter-leaflet.js

```diff
@@ -128,6 +138,32 @@ var LeafletMarkerView = LeafletUILayerView.extend({
         });
     },
 
+    remove: function() {
+        LeafletMarkerView.__super__.remove.apply(this, arguments);
+        var that = this;
+        this.icon_promise.then(function() {
+            if (that.icon) {
+                that.icon.remove();
+            }
+        });
+    },
+
+    set_icon: function(value) {
+        if (this.icon) {
+            this.icon.remove();
+        }
+        if (value) {
+            var that = this;
+            this.icon_promise = this.icon_promise.then(function() {
+                return that.create_child_view(value).then(function(view) {
+                    that.obj.setIcon(view.obj);
+                    that.icon = view;
+                    that.trigger('icon:created');
```

Comment:
I'm not sure I understand this line, why do you trigger this event? Is it used somewhere else?